stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/repository/MapperExtensions.kt

```diff
@@ -0,0 +1,168 @@
+    reactionCounts = reactionCounts.toMutableMap(),
+    reactionScores = reactionScores.toMutableMap(),
+    syncStatus = syncStatus,
+    shadowed = shadowed,
+    latestReactions = (latestReactions.map { it.toModel(getUser) }).toMutableList(),
+    ownReactions = (ownReactions.map { it.toModel(getUser) }).toMutableList(),
+    mentionedUsers = mentionedUsersId.map { getUser(it) }.toMutableList(),
+    replyTo = replyToId?.let { getMessage(it) }
+)
+
+internal fun Message.toEntity(): MessageEntity = MessageEntity(
+    id = id,
+    cid = cid,
+    userId = user.id,
+    text = text,
+    attachments = attachments,
+    syncStatus = syncStatus,
+    type = type,
+    replyCount = replyCount,
+    createdAt = createdAt,
+    createdLocallyAt = createdLocallyAt,
+    updatedAt = updatedAt,
+    updatedLocallyAt = updatedLocallyAt,
+    deletedAt = deletedAt,
+    parentId = parentId,
+    command = command,
+    extraData = extraData,
+    reactionCounts = reactionCounts,
+    reactionScores = reactionScores,
+    shadowed = shadowed,
+    latestReactions = latestReactions.map(Reaction::toEntity),
+    ownReactions = ownReactions.map(Reaction::toEntity),
+    mentionedUsersId = mentionedUsers.map(User::id),
+    replyToId = replyTo?.id
+)
+
+internal fun ChannelUserRead.toEntity(): ChannelUserReadEntity = ChannelUserReadEntity(getUserId(), lastRead)
+
+internal suspend fun ChannelUserReadEntity.toModel(getUser: suspend (userId: String) -> User): ChannelUserRead =
+    ChannelUserRead(getUser(userId), lastRead)
+
+internal fun Member.toEntity(): MemberEntity = MemberEntity(
+    userId = getUserId(),
+    role = role ?: user.role,
+    createdAt = createdAt,
+    updatedAt = updatedAt,
+    isInvited = isInvited ?: false,
+    inviteAcceptedAt = inviteAcceptedAt,
+    inviteRejectedAt = inviteRejectedAt,
+    shadowBanned = shadowBanned,
+)
+
+internal suspend fun MemberEntity.toModel(getUser: suspend (userId: String) -> User): Member = Member(
+    getUser(userId),
+    role,
+    createdAt,
+    updatedAt,
+    isInvited,
+    inviteAcceptedAt,
+    inviteRejectedAt,
+    shadowBanned,
+)
+
+internal fun Channel.toEntity(): ChannelEntity {
+    var lastMessage: MessageEntity? = null
+    var lastMessageAt: Date? = null
+    messages.lastOrNull()?.let { message ->
+        lastMessage = message.toEntity()
+        lastMessageAt = message.createdAt
+    }
+    return ChannelEntity(
+        type = type,
+        channelId = id,
+        cooldown = cooldown,
+        frozen = frozen,
+        createdAt = createdAt,
+        updatedAt = updatedAt,
+        deletedAt = deletedAt,
+        extraData = extraData,
+        syncStatus = syncStatus,
+        hidden = hidden,
+        hideMessagesBefore = hiddenMessagesBefore,
+        members = members.map(Member::toEntity).associateBy(MemberEntity::userId).toMutableMap(),
+        reads = read.map(ChannelUserRead::toEntity).associateBy(ChannelUserReadEntity::userId).toMutableMap(),
```

Comment:
There are some parts of our code where this collection is updated.
I would like to have immutable data here, but changing all properties to immutable data will do this PR bigger. It will be covered later. Our "Repository" design needs to be updated and doesn't allow to use of Entities out of the "Data Layer"